pkg/aws/eni/node.go

```diff
@@ -451,18 +591,22 @@ func (n *Node) CreateInterface(ctx context.Context, allocation *ipam.AllocationA
 		"addresses":         toAllocate,
 		"isPrefixDelegated": isPrefixDelegated,
 	})
+
 	scopedLog.Info("No more IPs available, creating new ENI")
 
-	eniID, eni, err := n.manager.api.CreateNetworkInterface(ctx, int32(toAllocate), subnet.ID, desc, securityGroupIDs, isPrefixDelegated)
+	eniID, eni, err := n.manager.api.CreateNetworkInterface(ctx, int32(toAllocate), subnet.ID, desc, securityGroupIDs, isPrefixDelegated, family)
 	if err != nil {
+		if family == ipam.IPv6 {
+			return 0, unableToCreateENI, fmt.Errorf("%s: %w", errUnableToCreateENI, err)
+		}
 		if isPrefixDelegated && isSubnetAtPrefixCapacity(err) {
-			// Subnet might be out of available /28 prefixes, but /32 IP addresses might be available.
+			// Subnet might be out of available /28 IPv4 prefixes, but /32 IP addresses might be available.
 			// We should attempt to allocate /32 IPs.
-			scopedLog.WithField(logfields.Node, n.k8sObj.Name).Warning("Subnet might be out of prefixes, Cilium will not allocate prefixes on this node anymore")
-			eniID, eni, err = n.manager.api.CreateNetworkInterface(ctx, int32(toAllocate), subnet.ID, desc, securityGroupIDs, false)
-		}
-		if err != nil {
-			return 0, unableToCreateENI, fmt.Errorf("%s: %w", errUnableToCreateENI, err)
+			scopedLog.WithField(logfields.Node, n.k8sObj.Name).Warning("Subnet might be out of IPv4 prefixes, Cilium will not allocate prefixes on this node anymore")
+			eniID, eni, err = n.manager.api.CreateNetworkInterface(ctx, int32(toAllocate), subnet.ID, desc, securityGroupIDs, false, family)
+			if err != nil {
```

Comment:
Moving this error handling inside the `isSubnetAtPrefixCapacity(err)` block here removes the `unableToCreateENI` error handling for other cases where `err != nil` on line 598.

It should remain outside this block.